src/circshiftedarray.jl

```diff
@@ -45,44 +45,50 @@ CircShiftedVector(v::AbstractVector, n = (0,)) = CircShiftedArray(v, n)
 
 size(s::CircShiftedArray) = size(parent(s))
 
+
 @inline function bringwithin(idx::Int, range::AbstractRange)
-    a, b = extrema(range)
-    n = length(range)
-    while idx < a
-        idx += n
-    end
-    while idx > b
-        idx -= n
-    end
-    idx
+    t = mod(idx - first(range), last(range))
+    return first(range) + t 
 end
 
-@inline bringwithin(idxs::Tuple, ranges::Tuple) =
-    (bringwithin(idxs[1], ranges[1]), bringwithin(Base.tail(idxs), Base.tail(ranges))...)
+@inline bringwithin(idxs::Tuple, ranges::Tuple) = bringwithin.(idxs, ranges)
```

Comment:
Probably better to leave as is, or just do `map(bringwithin, idxs, ranges)` and remove the method below. I think `map` is a bit more idiomatic for `Tuple`s than `broadcast`.